restricted_site_access.php

```diff
@@ -58,6 +58,50 @@ public static function _add_actions() {
 		add_action( 'wpmu_new_blog', array( __CLASS__, 'set_defaults' ), 10, 6 );
 		add_action( 'admin_enqueue_scripts', array( __CLASS__, 'enqueue_admin_script' ) );
 		add_action( 'wp_ajax_rsa_notice_dismiss', array( __CLASS__, 'ajax_notice_dismiss' ) );
+
+		// Filter the public option before saving
+		add_filter( 'pre_update_option_blog_public', array( __CLASS__, 'filter_blog_public_option_before_save' ), 10, 2 );
+		add_filter( 'pre_update_site_option_blog_public', array( __CLASS__, 'filter_blog_public_option_before_save' ), 10, 2 );
+		// Filter the public_option when we retrieve it so we can use existing logic
+		add_filter( 'option_blog_public', array( __CLASS__, 'filter_blog_public_after_retrieval' ) );
+		add_filter( 'site_option_blog_public', array( __CLASS__, 'filter_blog_public_after_retrieval' ) );
+	}
+
+
+	/**
+	 * Filter the value being saved to allow us to indicate if RSA is being used.
+	 *
+	 * @param $new_value
+	 * @param $old_value
+	 *
+	 * @return mixed
+	 */
+	public static function filter_blog_public_option_before_save( $new_value, $old_value ) {
+
+		if ( 2 === absint( $new_value ) ) {
+			// Set the actual value to 0 so we don't break get_sites()
+			update_option( 'rsa_active', 1 );
+			// Set the blog to not be indexed.
+			$new_value = 0;
```

Comment:
This should be set to 1 so that the site does not go into no-index if the plugin is removed or disabled